pop-cli-for-appchains/guides/test-runtime-upgrades.md

```diff
@@ -0,0 +1,157 @@
+---
+description: The following guide shows how to test runtime upgrades.
+---
+
+# Testing Runtime Upgrades
+
+A key feature of Substrate is its support for forkless upgrades. Testing the blockchain runtime upgrade process is essential to ensure a seamless network transition without disruptions.
+
+To simulate and validate the process of upgrading a blockchain's runtime, the `on-runtime-upgrade` executes the [`OnRuntimeUpgrade`](https://paritytech.github.io/polkadot-sdk/master/frame_support/traits/trait.OnRuntimeUpgrade.html) hooks of a runtime against the state of a live blockchain or a snapshot.
+
+Hence, there are two subcommands `live` and `snap` to specify the source of the runtime state.
+
+```bash
+Usage: pop test on-runtime-upgrade [OPTIONS] [COMMAND]
+
+Commands:
+  live  A live chain
+  snap  A state snapshot
+  help  Print this message or the help of the given subcommand(s)
+```
+
+## Test migrations
+
+By running the command `pop test on-runtime-upgrade`, you can test the [Runtime Upgrades](https://docs.polkadot.com/develop/parachains/maintenance/runtime-upgrades/) and [Storage Migrations](https://docs.polkadot.com/develop/parachains/maintenance/storage-migrations/) in a simulated environment.
+
```

Comment:
Jus t a suggestion to add here, but up to you:

Execute the following command to start the interactive deployment process:
```
pop test on-runtime-upgrade
```